cube/src/main/java/zone/cogni/asquare/cube/index/swap/IndexSwapService.java

```diff
@@ -0,0 +1,134 @@
+package zone.cogni.asquare.cube.index.swap;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+import zone.cogni.asquare.service.elasticsearch.ElasticStore;
+import zone.cogni.asquare.service.elasticsearch.info.ElasticsearchMetadata;
+import zone.cogni.asquare.service.elasticsearch.info.ElasticsearchMetadataService;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static zone.cogni.asquare.cube.index.swap.JsonBuilder.array;
+import static zone.cogni.asquare.cube.index.swap.JsonBuilder.object;
+
+public class IndexSwapService {
+
+  private static final Logger log = LoggerFactory.getLogger(IndexSwapService.class);
+
+  private final ElasticStore elasticStore;
+  private final ElasticsearchMetadataService elasticsearchMetadataService;
+
+  private final RestTemplate restTemplate = new RestTemplate(clientHttpRequestFactory(3000, 10000));
```

Comment:
can we come up with some way of having this in a default config that we can overwrite per project

Reply:
true. I am definitely not the HTTP in java expert. @domivds what is best?